### DISCUSSION TITLE
Node Without Input

### DISCUSSION THREAD

**Discussion by HakanSeven12** [leon-thomm/Ryven#190] [Q&A]
How can I create a node without an input?

When I add an exec input it works.

`
class PointZero_Node(GpNodeBase):

    title = 'Point0'

    init_inputs = [NodeInputType(type_='exec')]
    
    init_outputs = [
        NodeOutputType(),
    ]

    def place_event(self):
        point = gp_Pnt(0,0,0)
        self.set_output_val(0, OCCData(point))
`

But when I left init_inputs empty I cant use it

`
class PointZero_Node(GpNodeBase):

    title = 'Point0'

    init_inputs = []
    
    init_outputs = [
        NodeOutputType(),
    ]

    def place_event(self):
        point = gp_Pnt(0,0,0)
        self.set_output_val(0, OCCData(point))
`

**Comment by HakanSeven12** [comment]
@leon-thomm  can you help?

**Comment by Robsel** [comment]
It needs an Input to be able to be called.
The node doesn't exist without an Input, since the Inputs can activate whatever is inside the node. If you are looking for a 'start' node, it still needs to start somewhere and that is also its Input.
Or thats how I explained it, since you can run ryven without a gui there needs to be a Input to every node to start anything.

**Reply by leon-thomm** [reply]
Depends how you want to use it, you could also add some method to your node which you directly invoke in headless mode, but I agree the canonical way should be simply and input, which can be invoked manually but also connected to another node.
